fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java

```diff
@@ -195,6 +197,45 @@ public PhysicalPlanTranslator(PlanTranslatorContext context, StatsErrorEstimator
         this.statsErrorEstimator = statsErrorEstimator;
     }
 
+    // We use two phase read to optimize sql like: select * from tbl [where xxx = ???] [order by column1] [limit n]
+    // in the first phase, we add an extra column `RowId` to Block, and sort blocks in TopN nodes
+    // in the second phase, we have n rows, we do a fetch rpc to get all rowids data for the n rows
+    // and reconconstruct the final block
+    private void setResultSinkFetchOptionIfNeed() {
```

Comment:
do not add any optimize rules in translator. it is tricky and hard to maintain

Reply:
As we discussed, let's make these changes first, and subsequently support general lazy materialization.